editor/py/serve.py

```diff
@@ -151,12 +121,9 @@ def enumerate_dataset():
     """
     data = flask.request.get_json(force=True)
     basename, suffix = os.path.splitext(data['spreadsheet_name'])
-    temp_dir = get_user_path()
-    with tempfile.NamedTemporaryFile(mode='w', suffix=suffix,
-                                     dir=temp_dir) as f:
-        f.write(data['spreadsheet_data'].lstrip('ï»¿'))
-        f.seek(0)
-        dataframe = dataset_templating.read_spreadsheet(f.name)
+    spreadsheet_data = io.StringIO(data['spreadsheet_data'].lstrip('ï»¿'))
+    dataframe = dataset_templating.read_spreadsheet(spreadsheet_data,
```

Comment:
Did you test this at least through the GUI (if not an automated test)?

Reply:
It's tested programmatically in #327 and I tested it in the GUI on my `editor-tests` branch that combines #326 and #327.